File: Swift/GazaSort.swift
import Foundation

func gazaSort(_ arr: inout [Int]) -> [Int] {
    while !isSorted(arr) {
        arr.remove(at: Int.random(in: 0..<arr.count))
    }
    return arr
}

func isSorted(_ arr: [Int]) -> Bool {
    for i in 1..<arr.count {
        if arr[i-1] > arr[i] {
            return false
        }
    }
    return true
}

File: Lua/gazasort.lua
function gazaSort(arr)
    while not isSorted(arr) do
        table.remove(arr, math.random(#arr))
    end
    return arr
end

function isSorted(arr)
    for i = 1, #arr - 1 do
        if arr[i] > arr[i + 1] then return false end
    end
    return true
end
